lm75bd/lm75bd.c

```diff
@@ -26,8 +26,27 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
 }
 
 error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
-  /* Implement this driver function */
+  uint8_t registerPointer = 0x00;
+  uint8_t writeBuf[1] = {registerPointer};
   
+  error_code_t writeResult = i2cSendTo(devAddr, writeBuf, sizeof(writeBuf));
+  if (writeResult != ERR_CODE_SUCCESS) {
+    return writeResult;
+  }
+
+  uint8_t readBuf[2];
+  error_code_t readResult = i2cReceiveFrom(devAddr, readBuf, sizeof(readBuf));
+  if (readResult != ERR_CODE_SUCCESS) {
+    return readResult;
+  }
+
+  uint16_t tempBit = (readBuf[0] << 8) | readBuf[1];
+  if (tempBit & (1 << 15)) {
+    *temp = (int16_t)((tempBit >> 5) | 0xF800)*0.125;
+  } else {
+    *temp = (tempBit >> 5)*0.125;
+  }
```

Comment:
This code could just be
```suggestion 
*temp=(tempBit>>5)*0.125; 
```
If tempBit is a int16_t